src/elements/content-sidebar/ActivitySidebar.js

```diff
@@ -827,6 +875,149 @@ class ActivitySidebar extends React.PureComponent<Props, State> {
+                return;
+            }
+
+            // Check if the active entry is a first level Feed item
+            const firstLevelItem = this.getFeedItemById(feedItems, activeFeedEntryId);
+            if (firstLevelItem) {
+                const { id, type } = firstLevelItem;
+                if (
+                    type === FEED_ITEM_TYPE_ANNOTATION ||
+                    type === FEED_ITEM_TYPE_COMMENT ||
+                    type === FEED_ITEM_TYPE_TASK
+                ) {
+                    resolve({ id, type });
+                    return;
+                }
+                resolve({ id: activeFeedEntryId, type: activeFeedEntryType });
+                return;
+            }
+
+            // Check if the active entry is within replies of any first level Feed items
+            const firstLevelItemWithActiveReply = this.getFeedItemByReplyId(feedItems, activeFeedEntryId);
+            if (firstLevelItemWithActiveReply) {
+                const { id, type } = firstLevelItemWithActiveReply;
+                if (type === FEED_ITEM_TYPE_ANNOTATION || type === FEED_ITEM_TYPE_COMMENT) {
+                    resolve({ id, type });
+                    return;
+                }
+                resolve({ id: activeFeedEntryId, type: activeFeedEntryType });
+                return;
+            }
+
+            // If the active entry could not be found within feed items, it's most likely a reply that
+            // is not yet visible in feed and we need to fetch its data in order to find parent
+            this.getComment(
+                activeFeedEntryId,
+                ({ parent }) => {
+                    const parentItem = feedItems.find(({ id }) => id === parent?.id);
+                    if (
+                        !parentItem ||
+                        (parentItem.type !== FEED_ITEM_TYPE_ANNOTATION && parentItem.type !== FEED_ITEM_TYPE_COMMENT)
+                    ) {
+                        resolve({ id: activeFeedEntryId, type: activeFeedEntryType });
+                        return;
+                    }
+                    resolve({
+                        id: parentItem.id,
+                        type: parentItem.type,
+                    });
+                },
+                (error: ElementsXhrError) => {
+                    // If parent item not found resolve with values from props
+                    if (error.status === 404) {
+                        resolve({ id: activeFeedEntryId, type: activeFeedEntryType });
+                    } else {
+                        reject(error);
+                    }
+                },
+            );
+        });
+    };
+
+    /**
+     * Fetches comment data
+     *
+     * @param {string} id - id of the comment
+     * @param {Function} onSuccess - the success callback
+     * @param {ErrorCallback} onError -the error callback
+     * @return {void}
+     */
+    getComment = (
```

Comment:
Minor, but is this needed? Can we call `fetchThreadedComment` directly, instead?